utils/src/main/scala/za/co/absa/enceladus/utils/validation/field/FieldValidatorScalar.scala

```diff
@@ -24,43 +24,47 @@ import scala.util.control.NonFatal
 /**
   * Scalar types schema validation against default value
   */
-class FieldValidatorScalar extends FieldValidator {
+object FieldValidatorScalar extends FieldValidator {
   override def validateStructField(field: StructField): Seq[ValidationIssue] = {
-    val issues = new ListBuffer[ValidationIssue]
     try {
       val defaultName = "default"
       if (field.metadata contains defaultName) {
-        val default = field.metadata.getString("default")
+        val default = field.metadata.getString(defaultName)
         field.dataType match {
-          case v: BooleanType => default.toBoolean
-          case v: ByteType => default.toByte
-          case v: ShortType => default.toShort
-          case v: IntegerType => default.toInt
-          case v: LongType => default.toLong
-          case v: FloatType =>
+          case _: BooleanType => default.toBoolean
+          case _: ByteType => default.toByte
+          case _: ShortType => default.toShort
+          case _: IntegerType => default.toInt
+          case _: LongType => default.toLong
+          case _: FloatType =>
             val a: Float = default.toFloat
-            if (a == Float.NaN)
+            if (a == Float.NaN) {
```

Comment:
This is a tricky one. According to `IEEE574`, `NaN` is not a single value but quite a large set of values. The `Float.NaN` probably refers only to one of them, and the most likely this will work for the numeric string parsing. But to be really perfect you can use `Float.isNan()` or `Double.isNan()` instead, like in https://stackoverflow.com/questions/1456566/how-do-you-test-to-see-if-a-double-is-equal-to-nan.